File: android_app/app/src/main/java/com/oceansentinels/app/data/remote/api/IndianWeatherApi.kt
package com.oceansentinels.app.data.remote.api

import com.oceansentinels.app.data.remote.dto.IndianCityWeatherResponse
import com.oceansentinels.app.data.remote.dto.IndianGlobalWeatherResponse
import com.oceansentinels.app.data.remote.dto.IndianGlobalCurrentResponse
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Query

/**
 * IndianAPI.in Weather API
 * Base URL: https://weather.indianapi.in
 * Docs: https://indianapi.in/documentation/weather-api
 */
interface IndianWeatherApi {

    /**
     * Get weather for an Indian city (IMD data)
     * Fuzzy matching on city name
     */
    @GET("india/weather")
    suspend fun getIndianCityWeather(
        @Header("x-api-key") apiKey: String,
        @Query("city") city: String
    ): IndianCityWeatherResponse

    /**
     * Get current weather for any global location
     */
    @GET("global/current")
    suspend fun getGlobalCurrentWeather(
        @Header("x-api-key") apiKey: String,
        @Query("location") location: String
    ): IndianGlobalCurrentResponse

    /**
     * Get weather + forecast for a global location
     */
    @GET("global/weather")
    suspend fun getGlobalWeather(
        @Header("x-api-key") apiKey: String,
        @Query("location") location: String,
        @Query("days") days: Int = 3
    ): IndianGlobalWeatherResponse
}

File: IOS/OceanSentinels/Data/Remote/API/IndianWeatherAPI.swift
import Foundation

// MARK: - IndianAPI.in Weather Service

/// Replaces Retrofit IndianWeatherApi interface.
/// Base URL: https://weather.indianapi.in
final class IndianWeatherAPIService {
    
    private let baseURL = "https://weather.indianapi.in"
    private let apiKey: String
    private let session: URLSession
    private let decoder = JSONDecoder()
    
    init(apiKey: String = AppConfig.indianApiKey) {
        self.apiKey = apiKey
        let config = URLSessionConfiguration.default
        config.timeoutIntervalForRequest = 15
        self.session = URLSession(configuration: config)
    }
    
    private func buildRequest(path: String, queryItems: [URLQueryItem]) -> URLRequest {
        var components = URLComponents(string: baseURL + path)!
        components.queryItems = queryItems
        var request = URLRequest(url: components.url!)
        request.setValue(apiKey, forHTTPHeaderField: "x-api-key")
        return request
    }
    
    // MARK: - Indian City Weather (IMD data)
    
    func getIndianCityWeather(city: String) async throws -> IndianCityWeatherResponse {
        let request = buildRequest(
            path: "/india/weather",
            queryItems: [URLQueryItem(name: "city", value: city)]
        )
        let (data, _) = try await session.data(for: request)
        return try decoder.decode(IndianCityWeatherResponse.self, from: data)
    }
    
    // MARK: - Global Current Weather
    
    func getGlobalCurrentWeather(location: String) async throws -> IndianGlobalCurrentResponse {
        let request = buildRequest(
            path: "/global/current",
            queryItems: [URLQueryItem(name: "location", value: location)]
        )
        let (data, _) = try await session.data(for: request)
        return try decoder.decode(IndianGlobalCurrentResponse.self, from: data)
    }
    
    // MARK: - Global Weather + Forecast
    
    func getGlobalWeather(location: String, days: Int = 3) async throws -> IndianGlobalWeatherResponse {
        let request = buildRequest(
            path: "/global/weather",
            queryItems: [
                URLQueryItem(name: "location", value: location),
                URLQueryItem(name: "days", value: "\(days)")
            ]
        )
        let (data, _) = try await session.data(for: request)
        return try decoder.decode(IndianGlobalWeatherResponse.self, from: data)
    }
}
